packages/cli/src/utils/logger-transport-file.ts

```diff
@@ -0,0 +1,117 @@
+import type { LogLevel, LogRecord, LogTransport } from './logger';
+import { appendToFileSync } from './filesystem';
+import { existsSync, readdirSync, unlinkSync } from 'node:fs';
+import { dirname, extname, join } from 'node:path';
+
+export interface FileTransportOptions {
+  filePath?: string;
+  level?: LogLevel;
+  maxFiles?: number;
+}
+
+export class FileTransport implements LogTransport {
+  public filePath: string;
+  public level: LogLevel;
+  private maxFiles?: number;
+  private hasPruned = false;
+
+  constructor(options?: FileTransportOptions) {
+    this.filePath = options?.filePath ?? `./${Date.now()}.jsonl`;
+    this.level = options?.level ?? 'info';
+    this.maxFiles = options?.maxFiles;
+  }
+
+  public log(record: LogRecord): void {
+    if (!this.shouldLog(record.level)) {
+      return;
+    }
+
+    const line = this.format(record);
+    appendToFileSync(this.filePath, line);
+
+    if (!this.hasPruned && this.maxFiles !== undefined) {
+      // A log file is only created if logged at least once so we only prune old
+      // files if at least one log was written and not when initializing the
+      // transport.
+      this.hasPruned = true;
+      this.pruneOldFiles();
+    }
+  }
```

Comment:
question:

So if I understand correctly, pruning happens after the first log entry is written, right? Let's say we run a migration of 1000 entries, The remaining 999 log entries never trigger pruning again so you would end up with 999 until you do your next run.

Wondering if pruning should be a side effect of logging, but Im being very nit-picky about the single responsibility 🤔